### DISCUSSION TITLE
frigate on coral devboard and minidevboard

### DISCUSSION THREAD

**Discussion by saket424** [blakeblackshear/frigate#574] [Ask A Question] [answered]
Has anyone got frigate to work on either the coral devboard or the coral mini devboard running mendel ?
I just got this board and got docker working with it and now want to attempt to use frigate

@blakeblackshear what do i use for the detectors section ?
it is neither a pci nor a usb device so i am stuck. the parrot inference sample in the get-started section works fine

https://coral.ai/docs/dev-board-mini/get-started/
https://github.com/google-coral/edgetpu/issues/299

**Comment by blakeblackshear** [comment]
Did you try using `pci:0`? I think its using the same module, so that might work. If not, I would ask them what you would use for the dev mini for the device name as shown here: https://coral.ai/docs/edgetpu/multiple-edgetpu/#using-the-tensorflow-lite-python-api

**Reply by akostadinov** [reply]
do you think it is possible to use the coral dev board for detection only and run separately from the actual frigate server box?

**Comment by saket424** [comment]
@blakeblackshear 
Per your suggestion, I just now tried pci:0. No dice

 * Starting nginx nginx
   ...done.
frigate.mqtt                   INFO    : MQTT connected
detector.coral                 INFO    : Starting detection process: 35
frigate.edgetpu                INFO    : Attempting to load TPU as pci:0
frigate.app                    INFO    : Camera processor started for back: 38
frigate.app                    INFO    : Capture process started for back: 39
frigate.edgetpu                INFO    : No EdgeTPU detected. Falling back to CPU.

**Comment by saket424** [comment]
@blakeblackshear 
I tried usb:0 instead and it appears to have worked. but now i am getting some complaints from ffmpeg

 * Starting nginx nginx
   ...done.
frigate.mqtt                   INFO    : MQTT connected
detector.coral                 INFO    : Starting detection process: 34
frigate.app                    INFO    : Camera processor started for back: 37
frigate.edgetpu                INFO    : Attempting to load TPU as usb:0
frigate.app                    INFO    : Capture process started for back: 38
frigate.edgetpu                INFO    : TPU found
frigate.video                  INFO    : back: ffmpeg sent a broken frame. something is wrong.
frigate.video                  INFO    : back: ffmpeg process is not running. exiting capture thread...
frigate.video                  INFO    : back: ffmpeg sent a broken frame. something is wrong.
frigate.video                  INFO    : back: ffmpeg process is not running. exiting capture thread...
frigate.video                  INFO    : back: ffmpeg sent a broken frame. something is wrong.
frigate.video                  INFO    : back: ffmpeg process is not running. exiting capture thread...

I could have sworn that lsusb output wasnt Google earlier but now it is

```
mendel@vexing-bunny:~/IOTstack$ lsusb
Bus 002 Device 003: ID 18d1:9302 Google Inc. 
Bus 002 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
mendel@vexing-bunny:~/IOTstack$ 

```

**Comment by saket424** [comment]
The ffmpeg noise was relating to the hwaccel_flags. For the momemnt, I have commented it out.
I have frigate working now on the coral mini devboard. @blakeblackshear , any recommendations on what the
right hwaccel_flags showuld be?

**Comment by saket424** [comment, marked as answer]
It is a usb device. frigate worked with detector set as usb and TPU was found.

```
mendel@vexing-bunny:~/IOTstack$ lsusb
Bus 002 Device 003: ID 18d1:9302 Google Inc. 
Bus 002 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
mendel@vexing-bunny:~/IOTstack$ 
```

**Reply by akostadinov** [reply]
does it work in the long run? How many cameras could you connect and what is the power draw?
